src/Bicep.LangServer/Completions/BicepCompletionProvider.cs

```diff
@@ -168,6 +176,69 @@ private IEnumerable<CompletionItem> GetResourceTypeCompletions(SemanticModel mod
+        private IEnumerable<CompletionItem> GetModulePathCompletions(SemanticModel model, BicepCompletionContext context)
+        {
+            if (!context.Kind.HasFlag(BicepCompletionContextKind.ModulePath)
+            || FileResolver.TryResolveModulePath(model.SyntaxTree.FileUri, ".")?.LocalPath is not string cwd
+            || new Uri(cwd) is not {} cwdUri
+            || context.EnclosingDeclaration is not ModuleDeclarationSyntax declarationSyntax)
+            {
+                return Enumerable.Empty<CompletionItem>();
+            }
+
+            // To provide intellisense before the quotes are typed
+            if (declarationSyntax.Path is not StringSyntax stringSyntax 
+            || stringSyntax.TryGetLiteralValue() is not string entered)
+            {
+                entered = "";
+            }
+
+            if (FileResolver.TryResolveModulePath(cwdUri, entered) is not {} query
+            || FileResolver.FileExists(query))
+            {
+                return Enumerable.Empty<CompletionItem>();
+            }
+
+            var files = Enumerable.Empty<Uri>();
+            var dirs = Enumerable.Empty<Uri>();
+            var accesedDir = string.Empty;
+            if (FileResolver.DirExists(query))
+            {
+                accesedDir = entered;
+                files = FileResolver.GetFiles(query, "*.bicep");
+                dirs = FileResolver.GetDirectories(query, "");
+            }
+            else if (FileResolver.GetParentDirectory(query) is var parentDir
+                    && FileResolver.DirExists(parentDir))
+            {
+                accesedDir = entered.Remove(entered.Length - query.Segments[^1].Length);
+                files = FileResolver.GetFiles(parentDir, $"{query.Segments[^1]}*.bicep");
+                dirs = FileResolver.GetDirectories(parentDir, $"{query.Segments[^1]}*");
+            }
+            // "./" cannot be preserved when making relative Uris. We have to go and manually add it.
+            var fileItems = files
+                .Select(file => CreateModulePathCompletion(
+                    file.Segments.Last(), 
+                    (entered.StartsWith("./") ? "./" : "") + cwdUri.MakeRelativeUri(file).ToString(), 
+                    context.ReplacementRange, 
+                    CompletionItemKind.File, 
+                    CompletionPriority.High,
+                    false))
+                    .ToList();
+
+            // don't do completion range manipulation if we have to autocomplete entire path (module m <autocomplete>)
+            var dirItems = dirs
+                .Select(dir => CreateModulePathCompletion(
+                    dir.Segments.Last(), 
+                    (entered.StartsWith("./") ? "./" : "") + cwdUri.MakeRelativeUri(dir).ToString(),
+                    context.ReplacementRange, 
+                    CompletionItemKind.Folder, 
+                    CompletionPriority.Medium,
+                    declarationSyntax.Path is StringSyntax))
```

Comment:
Please add an explanation for this, or use an intermediary variable like `isTerminatedString` to make the reason for this clearer. (I'm guessing the purpose here is trying to detect when you're inside a string ending with `'`?)